test/publisher.js

```diff
@@ -35,6 +35,17 @@ describe('publisher', function () {
 
 			});
 
+            it( 'should return the proper address when a bucket name is provided', function () {
+                var bucketOptions = JSON.parse(JSON.stringify(options)); // to clone the object

```

Comment:
Either create a helper method to do the cloning (`close(options)` is clearer and doesn't require a comment) or make a second, appropriately named `options` object (global or local).